qemu/tests/watchdog.py

```diff
@@ -282,6 +283,45 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def heartbeat_test():
+        """
+        Heartbeat test for i6300esb
+        Test steps:
+        1.Start VM with "-watchdog-action pause" CLI option
+        2.Set heartbeat value and reload the i6300esb module
+        3.Trigger wathchdog action through open /dev/watchdog
+        4.Ensure watchdog_action takes effect after $heartbeat.
+        """
+
+        del_module_cmd = params["del_module_cmd"]
+        reload_module_cmd = params["reload_module_cmd"]
+        _watchdog_device_check(test, session, watchdog_device_type)
+
+        error_context.context("set heartbeat value and reload the i6300esb "
+                              "module", logging.info)
+        session.cmd(del_module_cmd)
+        heartbeat = random.randint(1, 180)
+        session.cmd_output(reload_module_cmd % heartbeat)
+        _watchdog_device_check(test, session, watchdog_device_type)
+
+        _trigger_watchdog(session, trigger_cmd)
+
+        error_context.context("Judge watchdog fire time", logging.info)
+        start_time = time.time()
+        end_time = start_time + float(heartbeat)
+        while not vm.monitor.verify_status("paused"):
+            if time.time() > end_time + 2:
+                test.fail("Monitor status is:%s, watchdog action '%s' didn't take"
+                          "effect" % (vm.monitor.get_status(), watchdog_action))
+            time.sleep(1)
+        guest_pause_time = time.time()
+        if time.time() >= end_time - 2:
```

Comment:
`if guest_pause_time >=end_time - 2`?  
Line 320 and line 323 you get time.time() for "Watchdog action came into effect after '%s's.", I guest that time should be `guest_pause_time` or `guest_pause_time - start_time` ?